src/kernel/src/crypto.rs

```diff
@@ -0,0 +1,63 @@
+use p256::ecdsa::{
+    signature::{self, Signer, Verifier},
+    Signature, SigningKey, VerifyingKey,
+};
+use sha2::{
+    digest::{
+        consts::{B0, B1},
+        generic_array::GenericArray,
+        typenum::{UInt, UTerm},
+    },
+    Digest, Sha256,
+};
+
+pub fn sha256(
+    input: impl AsRef<[u8]>,
+) -> GenericArray<u8, UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>> {
```

Comment:
Can you explain what's going on here with GenericArray?

Reply:
Yeah, basically GenericArray is just a way to support any slice primitive length of the returned value's 256 bytes, from [u8; 32] to [u128, 2]. I just edited that code to instead return [u8; 32] every time for simplicity.

Reply:
Great, I think this is perfect for now. If we want other slices and types we can add support later.